tests/integration_tests/performance/test_versioned_serialization_benchmark.py

```diff
@@ -0,0 +1,151 @@
+                "delta": 0.020  # milliseconds
+            }
+        },
+        "deserialize": {
+            "no-crc": {
+                "target": 0.034,  # milliseconds
+                "delta": 0.010  # milliseconds
+            },
+            "crc": {
+                "target": 0.042,  # milliseconds
+                "delta": 0.010  # milliseconds
+            }
+        }
+    },
+    "ARM": {
+        "serialize": {
+            "no-crc": {
+                "target": 0.096,  # milliseconds
+                "delta": 0.020  # milliseconds
+            },
+            "crc": {
+                "target": 0.186,  # milliseconds
+                "delta": 0.020  # milliseconds
+            }
+        },
+        "deserialize": {
+            "no-crc": {
+                "target": 0.034,  # milliseconds
+                "delta": 0.010  # milliseconds
+            },
+            "crc": {
+                "target": 0.042,  # milliseconds
+                "delta": 0.010  # milliseconds
+            }
+        }
+    }
+}
+
+
+def _check_statistics(directory, mean):
+    proc_model = [item for item in BASELINES if item in PROC_MODEL]
+    assert len(proc_model) == 1, "Could not get processor model!"
+
+    if "deserialize" in directory.lower():
+        bench = "deserialize"
+    else:
+        bench = "serialize"
+
+    if "crc" in directory.lower():
+        attribute = "crc"
+    else:
+        attribute = "no-crc"
+
+    measure = BASELINES[proc_model[0]][bench][attribute]
+    low = measure["target"] - measure["delta"]
+    high = measure["target"] + measure["delta"]
+    assert low <= mean <= high, "Benchmark result {} has changed!" \
+        .format(directory)
+
+
+@pytest.mark.skipif(
+    platform.machine() != "x86_64",
+    reason="Not supported yet."
+)
+def test_serialization_benchmark():
+    """Benchmark test for MicrovmState serialization/deserialization."""
+    logger = logging.getLogger("serialization_benchmark")
+
+    # Move into the benchmark directory
+    os.chdir(BENCHMARK_DIRECTORY)
+
+    # Run benchmark test
+    cmd = 'cargo bench'
+    result = utils.run_cmd_sync(cmd)
+    assert result.returncode == 0
+
+    # Parse each Criterion benchmark from the result folder and
+    # check the results against a baseline
+    results_dir = os.path.join(FC_WORKSPACE_DIR, "build/vmm_benchmark")
+    for directory in os.listdir(results_dir):
+        # Ignore the 'report' directory as it is of no use to us
+        if directory == "report":
+            continue
+
+        logger.info("Benchmark: %s", directory)
+
+        # Retrieve the 'estimates.json' file content
+        json_file = os.path.join(
+            results_dir,
+            "{}/{}".format(directory, "base/estimates.json"))
+        with open(json_file, "r") as read_file:
+            estimates = json.load(read_file)
+
+        # Save the Mean measurement(nanoseconds) and transform it(milliseconds)
+        mean = estimates['mean']['point_estimate'] / 1000000
```

Comment:
I think 1000000 should be a constant. Maybe something like: `NSEC_IN_MSEC = 1000`.

Reply:
Good idea